src/components/TextLink.tsx

```diff
@@ -0,0 +1,77 @@
+import React, {ForwardedRef, forwardRef, KeyboardEventHandler, MouseEventHandler, ReactElement} from 'react';
+import {GestureResponderEvent, Text as RNText, StyleProp, TextStyle} from 'react-native';
+import styles from '@styles/styles';
+import * as Link from '@userActions/Link';
+import CONST from '@src/CONST';
+import Text from './Text';
+
+type LinkProps = {
+    /** Link to open in new tab */
+    href: string;
+
+    onPress?: undefined;
+};
+
+type PressProps = {
+    href?: undefined;
+
+    /** Overwrites the default link behavior with a custom callback */
+    onPress: () => void;
+};
+
+type TextLinkProps = (LinkProps | PressProps) & {
+    /** Text content child */
+    children: ReactElement;
```

Comment:
You can use `ChildrenProps` here from `src/types/utils/ChildrenProps.ts`

Reply:
Actually after [this](https://github.com/Expensify/App/pull/30907#discussion_r1389592287) comment it is irrelevant. You can fix it directly in `TextProps` 👍 

```ts
type TextProps = RNTextProps &
    ChildrenProps & {
        /** The color of the text */
        color?: string;

        /** The size of the text */
        fontSize?: number;

        /** The alignment of the text */
        textAlign?: 'left' | 'right' | 'auto' | 'center' | 'justify';

        /** The family of the font to use */
        family?: keyof typeof fontFamily;
    };
```